--- id.go.orig	2015-04-23 17:33:40.000000000 -0500
+++ id.go	2015-04-23 17:34:22.000000000 -0500
@@ -53,14 +53,14 @@
 // Return the number of consecutive zeroes, starting from the low-order bit, in
 // a ID.
 func (id ID) PrefixLen() int {
-	for i := 0; i < IDBytes; i++ {
-		for j := 0; j < 8; j++ {
-			if (id[i]>>uint8(j))&0x1 != 0 {
-				return (8 * i) + j
-			}
-		}
-	}
-	return IDBytes * 8
+    for i := 0; i < IDBytes; i++ {
+        for j := 7; j >= 0; j-- {
+            if (id[i]>>uint8(j))&0x1 != 0 {
+                return (8 * i) + (7 - j)
+            }
+        }
+    }
+    return IDBytes * 8
 }
 
 // Generate a new ID from nothing.
